### PR TITLE
Feature/sqone 683/package check

### PR DESCRIPTION
**Pull request by dpellenwood**
## What does this do/fix?

* Updates the base Node & NPM to the latest LTS version. 
* Updates all NPM packages to the latest _supported_ version (`via yarn upgrade-interactive`)
* Minor tweaks to the WebPack config to reduce deprecation notices and dependencies.
* Updates deployments and CI actions to use noce version defined via NVM.

I tested builds after these updates w/o issue.  Note that these updates do _not_ bump any packages to newer versions that may break backwards compatibility. I'm going to try that in a spearate PR. :crossed_fingers: 

[SQONE-683]

[SQONE-683]: https://moderntribe.atlassian.net/browse/SQONE-683?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webpack/configs/app-base.js [hunk 1]
```diff
@@ -65,7 +65,6 @@ module.exports = {
 		],
 	},
 	optimization: {
-		emitOnErrors: false, // NoEmitOnErrorsPlugin
```
Comment (dpellenwood) [inline webpack/configs/app-base.js:68]: This is the default value defined by webpack, no need to make it explicit.

File: webpack/configs/dev-base.js [hunk 2]
```diff
@@ -27,7 +27,6 @@ module.exports = merge.strategy( {
 	],
 	optimization: {
 		splitChunks,
-		noEmitOnErrors: true, // NoEmitOnErrorsPlugin
```
Comment (dpellenwood) [inline webpack/configs/dev-base.js:30]: This optimization setting has been deprecated in favor of the reverse, which defaults to the setting we want: `emitOnErrors: false`.

File: webpack/optimization/minimizer.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 const TerserPlugin = require( 'terser-webpack-plugin' );
-const OptimizeCSSAssetsPlugin = require( 'optimize-css-assets-webpack-plugin' );
+const CssMinimizerPlugin = require( 'css-minimizer-webpack-plugin' );
```
Comment (dpellenwood) [inline webpack/optimization/minimizer.js:2]: `optimize-css-assets-webpack-plugin` is not recommended for webpack 5: https://www.npmjs.com/package/optimize-css-assets-webpack-plugin

**Review by jorgemasta — APPROVED** [review]
Looks nice 🚀

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -79,9 +79,11 @@ jobs:
         working-directory: dev/docker
 
       - name: Setup node
-        uses: actions/setup-node@v1
+        id: nvm
```
Comment (nickpelton) [inline .github/workflows/ci.yml:82]: ✨ Fancy

**Review by nickpelton — APPROVED** [review]

**Pull request merged**